dipy/reconst/ivim.py

```diff
@@ -622,6 +619,49 @@ def fit(self, data, bounds_de=None):
         result = np.insert(result, 0, np.mean(S0_est), axis=0)
         return IvimFit(self, result)
 
+    def _leastsq_vp(self, gtab, data, x_f):
+        """Use leastsq to find ivim_params
+
+        Parameters
+        ----------
+        gtab : GradientTable class instance
+        Gradient directions and bvalues.
+
+        data : array, (len(bvals))
+            An array containing the signal from a voxel.
+            If the data was a 3D image of 10x10x10 grid with 21 bvalues,
+            the multi_voxel decorator will run the single voxel fitting
+            on all the 1000 voxels to get the parameters in
+            IvimFit.model_paramters. The shape of the parameter array
+            will be (data[:-1], 4).
+
+        x_f : array
+              Initial guesses for the parameters S0, f, D_star and D
+              calculated using a linear fitting.
+
+        Returns
+        -------
+        x0 : array
+            Estimates of the parameters S0, f, D_star and D.
+        """
+        # bounds for nonlinear least squares fit
+        bounds = self.bounds
+
+        try:
+            res = least_squares(self.nlls_cost,
+                                x_f,
+                                bounds=bounds,
+                                args=(data,))
+            ivim_params = res.x
+            if np.all(np.isnan(ivim_params)):
+                return np.array([-1, -1, -1, -1])
+            return ivim_params
+
+        except ValueError:
+            ivimmodel = IvimModel(gtab, fit_method='trr')
```

Comment:
This quietly falls back to the `trr` method? I am not sure that's what we want here. Imagine a user comparing varpro to trr and getting exactly the same for both? Is there some way to instead throw a sufficiently informative error here?

Reply:
I understand what you mean, I could alternatively return the values from `x_f` obtained from `convex optimization + global optimization` of `VarPro` step too. 

But, this `x0` error typically occurs when the mask is not applied and is working in the background and not the data. 
So I feel that this should be okay.. Does that make sense?  

Nevertheless, I you feel that returning values from the previous step is better, I can do that too.. Just need to think of an appropriate way to do that.

Reply:
@Garyfallidis which one of the 2 alternatives do you suggest?